src/Microsoft.Identity.Web.Certificateless/AzureIdentityForKubernetesClientAssertion.Log.cs

```diff
@@ -0,0 +1,101 @@
+                if (logger != null && logger.IsEnabled(LogLevel.Information))
+                {
+                    __SignedAssertionEnvironmentVariableNotProvidedCallback(logger, environmentVariableName, null);
+                }
+            }
+            private static readonly Action<ILogger, Exception?> __NoSignedAssertionParameterProvidedCallback =
+                LoggerMessage.Define(LogLevel.Error, new EventId(3, nameof(NoSignedAssertionParameterProvided)), "The environment variable AZURE_FEDERATED_TOKEN_FILE or AZURE_ACCESS_TOKEN_FILE or the 'SignedAssertionFileDiskPath' must be set to the path of the file containing the signed assertion.");
+
+            public static void NoSignedAssertionParameterProvided(ILogger? logger)
+            {
+                if (logger != null && logger.IsEnabled(LogLevel.Error))
+                {
+                    __NoSignedAssertionParameterProvidedCallback(logger, null);
+                }
+            }
+            private static readonly Action<ILogger, string, Exception?> __FileAssertionPathNotFoundCallback =
+                LoggerMessage.Define<string>(LogLevel.Error, new EventId(4, nameof(FileAssertionPathNotFound)), "The file `{filePath}` containing the signed assertion was not found.");
+
+            public static void FileAssertionPathNotFound(ILogger? logger, string filePath)
+            {
+                if (logger != null && logger.IsEnabled(LogLevel.Error))
+                {
+                    __FileAssertionPathNotFoundCallback(logger, filePath, null);
+                }
+            }
+            private static readonly Action<ILogger, string, DateTime, Exception?> __SuccessFullyReadSignedAssertionCallback =
+                LoggerMessage.Define<string, DateTime>(LogLevel.Information, new EventId(5, nameof(SuccessFullyReadSignedAssertion)), "Successfully read the signed assertion for `{filePath}`. Expires at {expiry}.");
+
+            public static void SuccessFullyReadSignedAssertion(ILogger? logger, string filePath, DateTime expiry)
+            {
+                if (logger != null && logger.IsEnabled(LogLevel.Information))
+                {
+                    __SuccessFullyReadSignedAssertionCallback(logger, filePath, expiry, null);
+                }
+            }
+            private static readonly Action<ILogger, string, string, Exception?> __FileDoesNotContainValidAssertionCallback =
+                LoggerMessage.Define<string, string>(LogLevel.Error, new EventId(6, nameof(FileDoesNotContainValidAssertion)), "The file `{filePath} does not contain a valid signed assertion. {message}.");
+
+            public static void FileDoesNotContainValidAssertion(ILogger? logger, string filePath, string message)
+            {
+                if (logger != null && logger.IsEnabled(LogLevel.Error))
+                {
+                    __FileDoesNotContainValidAssertionCallback(logger, filePath, message, null);
+                }
+            }
+        }
```

Comment:
can you separate the private methods from the public ones. Here's an example: [https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapter.Logger.cs#L114](https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web.TokenCache/Distributed/MsalDistributedTokenCacheAdapter.Logger.cs#L114) 

and the public ones will need xml comments.

Reply:
Logger is a private class?